autoscale/src/main/java/com/sequenceiq/periscope/service/ScalingActivityService.java

```diff
@@ -56,29 +57,47 @@ public ScalingActivity findByCrn(String activityCrn) {
         return scalingActivityRepository.findByActivityCrn(activityCrn).orElseThrow(notFound("ScalingActivity", activityCrn));
     }
 
-    public List<ScalingActivity> findAllForCluster(Long clusterId) {
-        return scalingActivityRepository.findAllByCluster(clusterId);
+    public List<ScalingActivity> findAllForCluster(String clusterNameOrCrn, Pageable pageable) {
```

Comment:
Consider using the built-in `NameOrCrn` approach

Reply:
Done